src/item_factory.cpp

```diff
@@ -3223,48 +3251,28 @@ void Item_factory::load( islot_comestible &slot, const JsonObject &jo, const std
                                     jsobj.get_int( "probability" ) );
     }
 
-    bool is_not_boring = false;
     if( jo.has_member( "primary_material" ) ) {
-        std::string mat = jo.get_string( "primary_material" );
-        slot.specific_heat_solid = material_id( mat )->specific_heat_solid();
-        slot.specific_heat_liquid = material_id( mat )->specific_heat_liquid();
```

Comment:
Is the issue that eg. these materials may not be loaded yet?

Reply:
Yes.

Reply:
Oh and because it doesn't use generic factory we can't use that infra to resolve load order issues. Fun.